src/artwork.c

```diff
@@ -561,60 +542,99 @@ artwork_get_dir_image(struct evbuffer *evbuf, char *dir, int max_w, int max_h, c
 	  DPRINTF(E_SPAM, L_ART, "Trying directory artwork file %s\n", path);
 
 	  ret = access(path, F_OK);
-	  if (ret < 0)
-	    continue;
-
-	  // If artwork file exists (ret == 0), exit the loop
-	  break;
+	  if (ret == 0)
+	    {
+	      snprintf(out_path, PATH_MAX, "%s", path);
+	      return 0;
+	    }
 	}
+    }
 
-      // In case the previous loop exited early, we found an existing artwork file and exit the outer loop
-      if (j < nextensions)
-	break;
+  return -1;
+}
+
+static int
```

Comment:
I think the return from this function is unclear. There are ART_E_ERROR's, but also -1 and 0. From reading the code it looks to me like you should use ART_E, but maybe 0 and -1 can work as well. In any case, I suggest you document the function with input and output.